### DISCUSSION TITLE
Reuse only default frame

### DISCUSSION THREAD

**Discussion by asymmetric** [alphapapa/activities.el#70] [Ideas]
I use `activities-resume 'new` and it's great, but I think it would be even cooler if a further usecase could be supported: I always want activities to open a in a new frame (as I like to use my window manager for that), **unless** it's the default frame -- in which case I want `activities-resume` to behave as if `activities-resume-into-frame` was set to `current`.

This is because typically, I'll start Emacs and the first thing I'll do is resume an activity with `M-x activities-resume`, which then leaves the default frame open undefinitely, doing nothing.

A further improvement (which I'm not sure it's possible to implement) would be if the splash screen could prompt me about the activity I want to resume, and reuse that frame (which is the default one). Activities resumed via `M-x` would instead be resumed in new frames.

**Comment by alphapapa** [comment]
Hello,

> I use `activities-resume 'new` and it's great,

I guess you mean calling `activities-resume` with a universal prefix argument to reset the activity to its default state?  Otherwise I'm not sure what you mean.

>  but I think it would be even cooler if a further usecase could be supported: I always want activities to open a in a new frame (as I like to use my window manager for that), **unless** it's the default frame -- in which case I want `activities-resume` to behave as if `activities-resume-into-frame` was set to `current`.

I'm not sure what you mean by "default frame."  I'm aware of Emacs's concept of the "selected frame."
> 
> This is because typically, I'll start Emacs and the first thing I'll do is resume an activity with `M-x activities-resume`, which then leaves the default frame open undefinitely, doing nothing.

It sounds like the best option here might be to allow `activities-resume-into-frame` to be a function, which could be called to select the desired frame (including opening a new one).  Then you could write a function to do exactly what you want in every circumstance.

> 
> A further improvement (which I'm not sure it's possible to implement) would be if the splash screen could prompt me about the activity I want to resume, and reuse that frame (which is the default one). Activities resumed via `M-x` would instead be resumed in new frames.

If you mean the built-in Emacs splash screen, I've never looked into extending that, because I've never used it for anything.  I'm guessing that the `dashboard` package might be more appropriate to integrate with.

**Comment by jdtsmith** [comment]
Maybe they just mean having an interactive means of switching between the behaviors of `resume-into-frame=current`  and `resume-into-frame=new`?  I'd kind of like that too.  "Dedicate this frame I'm on to this activity" vs. "Create a new frame for this activity".

**Comment by asymmetric** [comment]
> I guess you mean calling `activities-resume` with a universal prefix argument to reset the activity to its default state? Otherwise I'm not sure what you mean.

@alphapapa I actually meant that I have `(setq activities-resume-into-frame 'new)` in my config, sorry for the confusion.

> I'm not sure what you mean by "default frame." I'm aware of Emacs's concept of the "selected frame."
> If you mean the built-in Emacs splash screen, I've never looked into extending that, because I've never used it for anything. I'm guessing that the `dashboard` package might be more appropriate to integrate with.

Indeed, I meant the frame that displays a splash screen. 


> It sounds like the best option here might be to allow `activities-resume-into-frame` to be a function, which could be called to select the desired frame (including opening a new one). Then you could write a function to do exactly what you want in every circumstance.

A combination of a custom function for `activities-resume-into-frame` and hooking this into the `dashboard` package does seem like what I want. I'll look into it as soon as I have more time, and report back with any progress.

Thanks for the help!
